devicetypes/fibargroup/fibaro-roller-shutter.src/fibaro-roller-shutter.groovy

```diff
@@ -0,0 +1,728 @@
+		} else if (!state.currentPreferencesState."$it.key".value) {
+			log.warn "Preference ${it.key} no. ${it.parameterNumber} has no value. Please check preference declaration for errors."
+		}
+	}
+	syncConfiguration()
+	// Preferences template end
+}
+
+private syncConfiguration() {
+	def commands = []
+	parameterMap.each {
+		try {
+			if (state.currentPreferencesState."$it.key".status == "syncPending") {
+				commands += encap(zwave.configurationV2.configurationSet(scaledConfigurationValue: getCommandValue(it), parameterNumber: it.parameterNumber, size: it.size))
+				commands += encap(zwave.configurationV2.configurationGet(parameterNumber: it.parameterNumber))
+			} else if (state.currentPreferencesState."$it.key".status == "disablePending") {
+				commands += encap(zwave.configurationV2.configurationSet(scaledConfigurationValue: it.disableValue, parameterNumber: it.parameterNumber, size: it.size))
+				commands += encap(zwave.configurationV2.configurationGet(parameterNumber: it.parameterNumber))
+			}
+		} catch (e) {
+			log.warn "There's been an issue with preference: ${it.key}"
+		}
+	}
+	sendHubCommand(commands)
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.configurationv2.ConfigurationReport cmd) {
+	// Preferences template begin
+	log.debug "Configuration report: ${cmd}"
+	def preference = parameterMap.find( {it.parameterNumber == cmd.parameterNumber} )
+	def key = preference.key
+	def preferenceValue = getPreferenceValue(preference, cmd.scaledConfigurationValue)
+	if (settings."$key" == preferenceValue) {
+		state.currentPreferencesState."$key".value = settings."$key"
+		state.currentPreferencesState."$key".status = "synced"
+		handleConfigurationChange(cmd)
+	} else if (preference.type == "boolRange") {
+		if (state.currentPreferencesState."$key".status == "disablePending" && preferenceValue == preference.disableValue) {
+			state.currentPreferencesState."$key".status = "disabled"
+		} else {
+			runIn(5, "syncConfiguration", [overwrite: true])
+		}
+	} else {
+		state.currentPreferencesState."$key"?.status = "syncPending"
+		runIn(5, "syncConfiguration", [overwrite: true])
+	}
+	// Preferences template end
+	handleConfigurationChange(cmd)
+}
+
+private getPreferenceValue(preference, value = "default") {
+	def integerValue = value == "default" ? preference.defaultValue : value.intValue()
+	switch (preference.type) {
+		case "enum":
+			return String.valueOf(integerValue)
+		case "boolean":
+			return String.valueOf(preference.optionActive == integerValue)
+		default:
+			return integerValue
+	}
+}
+
+private getCommandValue(preference) {
+	def parameterKey = preference.key
+	switch (preference.type) {
+		case "boolean":
+			return settings."$parameterKey" ? preference.optionActive : preference.optionInactive
+		case "boolRange":
+			def parameterKeyBoolean = parameterKey + "Boolean"
+			return !notNullCheck(settings."$parameterKeyBoolean") || settings."$parameterKeyBoolean" ? settings."$parameterKey" : preference.disableValue
```

Comment:
This code seems a bit strange... Was there a reason for not just doing something like  `(settings."$parameterKeyBoolean" == null)`?

If you do need to keep it, I'd recommend using something like `isNull` and `isNotNull`.

Reply:
I'd like to believe there was a reason, in other case this seems to be utterly nonsense. Anyway, I've changed it to something normal.